ssl/ssl_lib.c

```diff
@@ -2277,39 +2277,47 @@ int SSL_get_key_update_type(const SSL *s)
     return s->key_update;
 }
 
-int SSL_renegotiate(SSL *s)
+/*
+ * Can we accept a renegotiation request?  Return 1 if yes. If not,
+ * set error and return 0.
+ */
+static int accept_renegotiation(const SSL *s)
 {
     if (SSL_IS_TLS13(s)) {
         ERR_raise(ERR_LIB_SSL, SSL_R_WRONG_SSL_VERSION);
         return 0;
     }
 
-    if ((s->options & SSL_OP_NO_RENEGOTIATION)) {
-        ERR_raise(ERR_LIB_SSL, SSL_R_NO_RENEGOTIATION);
+    if (SSL_is_server(s)
+            && (s->options & SSL_OP_ALLOW_CLIENT_RENEGOTIATION) == 0)
```

Comment:
This seems wrong. It appears to prevent server initiated renegotiation (on the server side) if the option is not set.

Reply:
Yes, this is in the wrong place. It should be, as you said, around line 1384 of `statem_server.c`

I'll address all your feedback by Monday; today's a holiday/wellness-day for work.